crates/router/src/core/payment_methods/vault.rs

```diff
@@ -0,0 +1,561 @@
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Error parsing Payment Method from Values")?;
+
+        Ok((Some(payment_method), customer_id))
+    }
+
+    #[instrument(skip_all)]
+    pub async fn store_payment_method_data_in_locker(
+        state: &routes::AppState,
+        token_id: Option<String>,
+        payment_method: &api::PaymentMethod,
+        customer_id: Option<String>,
+    ) -> RouterResult<String> {
+        let value1 = payment_method
+            .get_value1(customer_id.clone())
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Error getting Value1 for locker")?;
+
+        let value2 = payment_method
+            .get_value2(customer_id)
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Error getting Value12 for locker")?;
+
+        let lookup_key = token_id.unwrap_or_else(|| generate_id_with_default_len("token"));
+
+        create_tokenize(state, value1, Some(value2), lookup_key).await
+    }
+
+    #[instrument(skip_all)]
+    pub async fn delete_locker_payment_method_by_lookup_key(
+        state: &routes::AppState,
+        lookup_key: &Option<String>,
+    ) {
+        if let Some(lookup_key) = lookup_key {
+            let delete_resp = delete_tokenized_data(state, lookup_key).await;
+            match delete_resp {
+                Ok(resp) => {
+                    if resp == "Ok" {
+                        logger::info!("Card From locker deleted Successfully")
+                    } else {
+                        logger::error!("Error: Deleting Card From Locker : {}", resp)
+                    }
+                }
+                Err(err) => logger::error!("Err: Deleting Card From Locker : {}", err),
+            }
+        }
+    }
+}
+
+//------------------------------------------------TokenizeService------------------------------------------------
+pub async fn create_tokenize(
+    state: &routes::AppState,
+    value1: String,
+    value2: Option<String>,
+    lookup_key: String,
+) -> RouterResult<String> {
+    let payload_to_be_encrypted = api::TokenizePayloadRequest {
+        value1,
+        value2: value2.unwrap_or_default(),
+        lookup_key,
+        service_name: "CARD".to_string(),
+    };
+    let payload = serde_json::to_string(&payload_to_be_encrypted)
+        .map_err(|_x| errors::ApiErrorResponse::InternalServerError)?;
+    let encrypted_payload = services::encrypt_jwe(&state.conf.jwekey, &payload)
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Error getting Encrypt JWE response")?;
+
+    let create_tokenize_request = api::TokenizePayloadEncrypted {
+        payload: encrypted_payload,
+        key_id: services::get_key_id(&state.conf.jwekey).to_string(),
+        version: Some("0".to_string()),
```

Comment:
Hard coded values can we move to static.

Reply:
Done.